src/internal/connector/onedrive/restore.go

```diff
@@ -800,8 +805,37 @@ func AugmentRestorePaths(backupVersion int, paths []path.Path) ([]path.Path, err
 				break
 			}
 
-			colPaths[np.String()] = np
-			p = np
+			// For now only allow storage and restore paths where the restore path is
+			// at least as long as the storage path. The path munging we have now
+			// should only produce that. If the restore path is shorter than the
+			// storage path it's not clear how we should handle permissions. If the
+			// restore path is longer than the storage path then the first few
+			// (closest to the root) directories will have default permissions.
+			if len(p.RestorePath.Elements()) < len(np.Elements()) {
+				return nil, clues.New("RestorePath shorter than StoragePath").
+					With("restore_path", p.RestorePath, "storage_path", np)
+			}
+
+			pd := p.RestorePath
+
+			// Make sure the RestorePath always points to the level of the current
+			// collection. We need to track if it's the first iteration because the
+			// RestorePath starts out at the collection level to begin with.
+			if !first {
+				pd, err = p.RestorePath.Dir()
+				if err != nil {
+					return nil, err
+				}
+			}
```

Comment:
I can move the comment on line 808 up to the function definition, but I believe it discusses much of what you've asked about

Reply:
Yeah, wouldn't hurt.  The current phrasing is a little wishy washy.  Something simple like "precondition: restore paths must be at least as long as storage paths.  Longer restore paths will work fine, but this will only create dirmeta files for as many levels in the restore path as there are in the storage path."